### ISSUE TITLE
On restart of a persisted instance, getting "Failed to load instance test-instance from storage: DnaMissing"

### ISSUE THREAD

**Issue by Connoropolous** [holochain/holochain-rust#1768]
Then it goes to 
`Initializing new chain...`

I might be missing something, but I think this is a bug wherein you can't actually reload a whole chain from storage :s

**Comment by Connoropolous** [comment]
I think the problem is that it doesn't exit when it encounters this kind of error...
https://github.com/holochain/holochain-rust/blob/d83246669ba565ce659cb036a5353ab1fae96d02/conductor_api/src/conductor/base.rs#L880-L888

**Comment by Connoropolous** [comment]
@lucksus @AshantiMutinta @zippy can you provide thoughts on this?

**Comment by Connoropolous** [comment]
In general this makes sense cause I hear that it doesn't actually write the DNA into storage

**Comment by jamesray1** [comment]
May you please add more repro steps?

**Comment by thedavidmeister** [comment]
@jamesray1 yes you couldn't safely save any state that is only unique to a given process (e.g. as opposed to UUIDs) - that said i haven't looked around to see if there are still snowflakes in the state (those comments are pretty old)

**Comment by jamesray1** [comment]
LTM like there are still snowflakes, e.g.:

In the network state:
https://github.com/holochain/holochain-rust/blob/aa53657ee25bd35304c177a8d5560318644b0ccf/crates/core/src/network/state.rs#L68

In ActionWrapper
https://github.com/holochain/holochain-rust/blob/master/crates/core/src/action.rs#L40-L55

etc:
https://github.com/holochain/holochain-rust/search?q=snowflake&unscoped_q=snowflake

**Comment by thedavidmeister** [comment]
ok, if these are being saved/loaded it makes sense to me to use UUIDs instead
